File: API/Models/Post.cs
using System.ComponentModel.DataAnnotations;

namespace BlogAPI.Models
{
    public class Post
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;
        
        [Required]
        public string Content { get; set; } = string.Empty;
        
        public string? ImageUrl { get; set; }
        
        public int LikesCount { get; set; } = 0;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // Внешний ключ
        public int AuthorId { get; set; }
        
        // Навигационные свойства
        public User Author { get; set; } = null!;
        public ICollection<Like> Likes { get; set; } = new List<Like>();
    }
}

File: BlogAndroid/app/src/main/java/com/blog/android/models/Post.kt
package com.blog.android.models

data class Post(
    val id: Int,
    val title: String,
    val content: String,
    val imageUrl: String?,
    val likesCount: Int,
    val createdAt: String,
    val updatedAt: String,
    val author: User,
    val isLikedByCurrentUser: Boolean
)

data class CreatePost(
    val title: String,
    val content: String,
    val imageUrl: String?
)

data class UpdatePost(
    val title: String,
    val content: String,
    val imageUrl: String?
)
